### DISCUSSION TITLE
Calculation of average value by generation type

### DISCUSSION THREAD

**Discussion by CHGR11** [opennem/opennem#311] [Q&A]
Hi There

Thank you very much for developing OpenNEM, it's a great platform.

I was hoping you could provide me with an understanding of how the "Ave. Value per MWh" is calculated for each generation type.

Is it the production of power provided during each interval period and balancing price received for that interval?

Thanks!

![image](https://github.com/opennem/opennem/assets/137369926/1c5f22a7-db39-4578-859c-e139fa3da2ae)

**Comment by aleith** [comment]
its a weight average, so` 𝜮 pₜ · Eₜ / 𝜮 Eₜ ` where  **pₜ** is the price for trade interval **𝒕** and **Eₜ** is the energy traded in interval **𝒕**  for the technology in question.

put another way, its calculated by finding the sum (𝜮) of all the products of each trade-interval price (in $ / MWh) and the energy supply (in MWh) for the period in question (result in $ ), then dividing that result [sum product] by the sum of the energy traded for that technology category (in MWh) across the period in question. Answer is in $/MWh.
